migrations/0005_access.sql

```diff
@@ -0,0 +1,21 @@
+-- Migration number: 0005 	 2025-01-30T00:58:45.123Z
+
+DROP TABLE IF EXISTS access;
+
+-- Store authentication and authorization data separately from profiles
+CREATE TABLE IF NOT EXISTS access (
+	-- The UUID of the profile this access belongs to
+	id TEXT NOT NULL UNIQUE COLLATE BINARY,
+	-- Schema version to use
+	schemaVersion INTEGER NOT NULL DEFAULT 1,
+	-- The person's access level in the system
+	-- This determines what actions they can perform
+	access TEXT NOT NULL CHECK(access IN ('admin', 'organizer', 'volunteer')),
+	-- The date this access was created
+	happenedAt DATETIME NOT NULL,
+	-- The date this access was added to the database
+	insertedAt DATETIME NOT NULL,
+
+	PRIMARY KEY (id),
+	FOREIGN KEY (id) REFERENCES profile(id)
+);
```

Comment:
Using a primary key as a foreign key is not ideal because it enforces a strict one-to-one relationship, limiting flexibility. This prevents a profile from having multiple access roles.

I suggest using a separate `id` as the primary key while keeping `profile_id` as a foreign key.

```suggestion
CREATE TABLE IF NOT EXISTS access (
	-- Unique identifier for each access entry
	id TEXT NOT NULL UNIQUE COLLATE BINARY,
	-- The UUID of the profile this access belongs to
	profile_id TEXT NOT NULL UNIQUE COLLATE BINARY,
	-- Schema version to use
	schemaVersion INTEGER NOT NULL DEFAULT 1,
	-- The person's access level in the system
	-- This determines what actions they can perform
	access TEXT NOT NULL CHECK(access IN ('admin', 'organizer', 'volunteer')),
	-- The date this access was created, saved as an ISO timestamp
	happenedAt DATETIME NOT NULL,
	-- The date this access was added to the database, saved as an ISO timestamp
	insertedAt DATETIME NOT NULL,

	PRIMARY KEY (id),
	FOREIGN KEY (profile_id) REFERENCES profile(id)
);
```
For your understanding, you can also refer to the `0004_create-role.sql`